odoo/tools/convert.py

```diff
@@ -142,7 +142,7 @@ def _process(s):
             return '<?xml version="1.0"?>\n'\
                 +_process("".join(etree.tostring(n, encoding='unicode') for n in node))
         if t == 'html':
-            return _process("".join(etree.tostring(n, encoding='unicode') for n in node))
+            return _process("".join(etree.tostring(n, method='html', encoding='unicode') for n in node))
```

Comment:
I think Julien validated it, but would also be good to know if I can backport to 16.0 as there also is an issue with mail template 'reset' because of this (and because editor isn't consistent about using jquery or not, as outlined in my second comment on this PR)

Reply:
@xmo-odoo hey, re-launching you on this

As pointed out by xmo, what used to fix this is technically unsafe because expanding `<script><script/>` to `<script><script></script>` could pose some issues. Moreover using the sanitizer on this content anywhere would entirely break the templates.

By serializing it as html via lxml, we would ensure that the content is actually safe (having been parsed as valid xml first, we're sure there would not be any weird business like that).

So as it stands, I don't think there are many good and safe alternatives to the back-end commit.

Reply:
@xmo-odoo here^

I thought you were saying you didn't want to parse templates as HTML

Reply:
Well technically, bouncing off the alternative I mentionned in https://github.com/odoo/odoo/pull/114632#discussion_r1153269516, it might be enough for our use case to use `.innerHtml` rather than jQuery's `.html(` method. The reason I suggested to use `DOMPurify` in the first place was to be extra safe so that nobody would fall into the same trap again, but if the extra safety measure is not possible because `DOMPurify` would destroy the template then we could forego the extra safety altogether.

I'd say it's not that big a deal in the end because:

1. The only reason why you need this function is kind of an edge case already so I guess the probability of someone else needing to use this function is rather low.

2. We plan to remove jQuery and, unlike 8 years ago, we are actually working on it so it's closer than ever.